blocks/library/embed/index.js

```diff
@@ -118,17 +144,24 @@ registerBlockType( 'core/embed', {
 						} else {
 							this.setState( { error: true } );
 						}
-						this.setState( { fetching: false } );
+						this.setState( { fetching: false, loadingFromSavedBlock: false } );
 					} );
 				}
 			);
 		}
 
 		render() {
-			const { html, type, error, fetching } = this.state;
+			const { html, type, error, fetching, loadingFromSavedBlock } = this.state;
 			const { url, caption } = this.props.attributes;
 			const { setAttributes, focus, setFocus } = this.props;
 
+			if ( loadingFromSavedBlock ) {
+				// we're loading from a saved block, but haven't fetched the HTML yet...
+				return (
```

Comment:
@jasmussen what do you think should be the UX here? With this change it'd be the url as text shown until we can load the generated embed. Maybe a white box saying "Generating embed..." with the url in smaller text could be better?